docs/ProgrammingConcerns.md

```diff
@@ -0,0 +1,176 @@
+# Memory allocation routines
```

Comment:
Could you add a section to the document explaining which subprojects the guidelines apply to? E.g. external applications like FEXRootFSFetcher are obviously not bound to this, but for unit tests it's not so clear.

Some of the mentions of pecularities around thunking could be moved to such a section as well.

Reply:
Added a section